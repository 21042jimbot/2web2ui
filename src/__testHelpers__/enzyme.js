export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
