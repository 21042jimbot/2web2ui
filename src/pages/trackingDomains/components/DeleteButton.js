import React from 'react';
import { Button } from '@sparkpost/matchbox';

export default function({ status, onClick }) {
  if (status === 'pending' || status === 'blocked') {
    return null;
  }
  return <Button destructive size='small' onClick={onClick}>Delete</Button>;
}
