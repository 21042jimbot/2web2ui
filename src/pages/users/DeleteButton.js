import React, { Component } from 'react';
import { Button, Icon, Modal, Panel } from '@sparkpost/matchbox';

// Delete button with a confirmation modal
export default class DeleteButton extends Component {
  static defaultProps = {
    disabled: false
  }

  constructor(props) {
    super(props);
    this.state = { open: false };
  }

  onCancel = () => {
    this.setState({ open: false });
  }

  onSubmit = () => {
    this.setState({ open: false }, () => {
      const { username } = this.props.user;
      this.props.onSubmit(username);
    });
  }

  onOpen = () => {
    this.setState({ open: true });
  };

  render() {
    const { disabled, user } = this.props;

    // Display nothing when disabled to match current UI style
    if (disabled) { return null; }

    return (
      <span>
        <Button plain onClick={this.onOpen}>
          <Icon name="Delete"/>
        </Button>
        <Modal open={this.state.open}>
          <Panel accent sectioned title="Delete User">
            <p>Are you sure you want to delete {user.name}?</p>
            <Button primary onClick={this.onSubmit} style={{ marginRight: '1rem' }}>
              Delete
            </Button>
            <Button onClick={this.onCancel}>Cancel</Button>
          </Panel>
        </ Modal>
      </span>
    );
  }
}
