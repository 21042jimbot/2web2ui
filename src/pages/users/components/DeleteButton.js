import React from 'react';
import fp from 'lodash/fp';
import { Button, Icon } from '@sparkpost/matchbox';

export default function DeleteButton({ disabled = false, name = null, onClick = fp.noop }) {
  // Display nothing when disabled to match current UI style
  if (disabled) { return null; }

  const handleClick = (event) => { onClick(name); };

  return (
    <Button plain name={name} onClick={handleClick}>
      <Icon name="Delete"/>
    </Button>
  );
}
