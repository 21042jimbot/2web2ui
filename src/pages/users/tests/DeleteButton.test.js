import { shallow } from 'enzyme';
import fp from 'lodash/fp';
import React from 'react';

import DeleteButton from '../DeleteButton';

const defaultProps = {
  user: {},
};

const TestComponent = (props = {}) => {
  const nextProps = fp.merge(defaultProps, props);
  return shallow(<DeleteButton {...nextProps} />);
};

it('renders a button and hidden modal', () => {
  const wrapper = TestComponent();
  expect(wrapper).toMatchSnapshot();
});

it('renders nothing', () => {
  const wrapper = TestComponent({ disabled: true });
  expect(wrapper).toMatchSnapshot();
});
