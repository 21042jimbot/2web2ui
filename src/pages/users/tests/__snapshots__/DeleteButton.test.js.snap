// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a button and hidden modal 1`] = `
<span>
  <t
    onClick={[Function]}
    plain={true}
    size="default"
  >
    <t
      name="Delete"
    />
  </t>
  <t
    open={false}
  >
    <t
      accent={true}
      sectioned={true}
      title="Delete User"
    >
      <p>
        Are you sure you want to delete 
        ?
      </p>
      <t
        onClick={[Function]}
        primary={true}
        size="default"
        style={
          Object {
            "marginRight": "1rem",
          }
        }
      >
        Delete
      </t>
      <t
        onClick={[Function]}
        size="default"
      >
        Cancel
      </t>
    </t>
  </t>
</span>
`;

exports[`renders nothing 1`] = `null`;
